runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -41,7 +41,7 @@ def get_long_running_operation_result(azure_response, custom_deserialization_blo
       if !AsyncOperationStatus.is_terminal_status(polling_state.status)
         task = Concurrent::TimerTask.new do
           begin
-            if !polling_state.azure_async_operation_header_link.nil?
+            if !polling_state.azure_async_operation_header_link.nil? && custom_deserialization_block.nil?
```

Comment:
@sarangan12 With the proposed fix if we have operation header and no location header and customer_deserialization_block is not nil (and is not a 'put'), we return AzureOperationError 'Location header is missing...', is that the correct result for this case? Is that case possible? Without the fix, in that case we would go in the first "if" clause, and execute different code, would that give a different result?